### PR TITLE
Add UI, unit tests, and Preserve data for Movie and Source

### PR DESCRIPTION
**Pull request by diarisdiakite**
In this PR, I added unit tests for all implemented methods (issue https://github.com/Rebelzob/catalog-of-my-things/issues/34):
📃 For the methods in the Source class
✔️ initialize source
✔️ add_a_source method
✔️ list_all_sources method
✔️ find_source_by_index
✔️ add_item
📃 For the methods in the Movie class
✔️ initialize
✔️ add_a_movie
✔️ find_movie_by_index
✔️ can_be_archived
📃 Added a UI for the Movie and Source options with the following ones available (issue https://github.com/Rebelzob/catalog-of-my-things/issues/35)
✔️ List all movies
✔️ List all sources (e.g. 'From a friend', 'Online shop')
✔️ Add a movie
📃 Created a schema.sql file with tables that is analogical to the structure of the classes that I created (issue https://github.com/Rebelzob/catalog-of-my-things/issues/30, issue https://github.com/Rebelzob/catalog-of-my-things/issues/31 and Issue https://github.com/Rebelzob/catalog-of-my-things/issues/37)
✔️ movies table (add all properties and associations from the parent Item class as table columns)
✔️ sources table
📃 Created files to preserve_data files (issue #36)
✔️ created a save_data and a load_data for the movie class instances
✔️ created a save_data and a load_data files for the source class instances
✔️ save all the data in json files.

### CODE REVIEW

File: app/spec/movie_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require_relative '../models/movie'
+
+describe Movie do
+  context 'Given a provided movie' do
+    # params: title, source, publish_date, archived, silent
+    movie = Movie.new('Titanic', 'Youtube', '1997-12-31', true, true)
```
Comment (kambereBr) [inline app/spec/movie_spec.rb:6]: Once, Movie class refactored, please make sure to update parameters in this file as well. That way, all unit tests pass

**Review by kambereBr — CHANGES_REQUESTED** [review]
STATUS : CHANGE REQUIRED ♻️ 

Hey @diarisdiakite, 
Good job so far! Please check comments and try to apply if possible.

Happy coding !

File: app/models/movie.rb [hunk 2]
```diff
@@ -1,26 +1,41 @@
...
-  attr_accessor :name, :genre, :author, :source, :label, :publish_date, :archived, :silent
-
-  include ItemData
+  attr_accessor :title, :genre, :author, :source, :label, :publish_date, :archived, :silent
 
   @movies = []
 
-  def initialize(silent, *args)
-    super(*args)
+  def initialize(_title, _source, _publish_date, _archived, silent)
```
[the remark this answers is not in the document]
Comment (diarisdiakite) [reply app/models/movie.rb]: @kambereBr Updated. Thanks for the review.

File: app/spec/movie_spec.rb (hunk 1 above)
Comment (diarisdiakite) [reply app/spec/movie_spec.rb]: Updated @kambereBr. Thanks for the review and for your help.

**Pull request merged**